Objects/longobject.c

```diff
@@ -26,15 +26,18 @@ class int "PyObject *" "&PyLong_Type"
 _Py_IDENTIFIER(little);
 _Py_IDENTIFIER(big);
 
+/* Is this PyLong of size 1, 0 or -1? */
+#define IS_MEDIUM_VALUE(x) (((size_t)Py_SIZE(x)) + 1 < 3)
+
 /* convert a PyLong of size 1, 0 or -1 to an sdigit */
-#define MEDIUM_VALUE(x) (assert(-1 <= Py_SIZE(x) && Py_SIZE(x) <= 1),   \
-         Py_SIZE(x) < 0 ? -(sdigit)(x)->ob_digit[0] :   \
-             (Py_SIZE(x) == 0 ? (sdigit)0 :                             \
-              (sdigit)(x)->ob_digit[0]))
+#define MEDIUM_VALUE(x) (assert(IS_MEDIUM_VALUE(x)), \
+    ((sdigit)(Py_SIZE(x) * (x)->ob_digit[0])))
```

Comment:
I like static inline functions, why not using them here? C compilers are now very efficient to inline and optimize them, as much as macros.